lnwallet/channel.go

```diff
@@ -377,6 +377,10 @@ type PaymentDescriptor struct {
 	// blinded route (ie, not the introduction node) from update_add_htlc's
 	// TLVs.
 	BlindingPoint lnwire.BlindingPointRecord
+
+	// WireRecordsBlob contains the TLV records blob that was included in
+	// the original wire message that added this HTLC.
+	WireRecordsBlob lnwire.CustomRecordsBlob
```

Comment:
So I have something of this logic in: https://github.com/lightningnetwork/lnd/pull/8562/commits/99db910f726a622202341fcc089755114adfdc7c

Later commits then ensure that this is threaded all the way down to disk. This takes an approach of just accepting the blob, instead of the wire record map.

Reply:
Should attempt to rebase over that. That also handles all the accounting for restart/retransmission logic not in this PR.